src/compiler/checker.ts

```diff
@@ -33236,7 +33236,6 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         const candidates = candidatesOutArray || [];
         // reorderCandidates fills up the candidates array directly
         reorderCandidates(signatures, candidates, callChainFlags);
-        Debug.assert(candidates.length, "Revert #54442 and add a testcase with whatever triggered this");

```

Comment:
Maybe at this point we just remove the message and leave the assert? Would this check here be generally helpful to prevent future bugs or aid debugging?